server/apiserver/argoserver.go

```diff
@@ -314,6 +316,15 @@ func (as *argoServer) newGRPCServer(instanceIDService instanceid.Service, offloa
 	return grpcServer
 }
 
+func customHandler(ctx context.Context, mux *runtime.ServeMux, marshaller runtime.Marshaler, w http.ResponseWriter, r *http.Request, err error) {
+	_, ok := status.FromError(err)
+	if !ok {
+		// InvalidArgument is translated to a BadRequest
```

Comment:
@isubasinghe this will convert all errors into badrequest. we need to send this error if any spec validation error.

Reply:
@sarabala1979 Do you know anyway to determine if an error is a spec validation error? I wasn't able to find how, hence I felt forced to go with this approach. The limitation is from the fact that some errors are returned by fmt.Errorf, which makes it impossible to check against since they are just strings. 

I think personally the fix is to create a new `ServerError` struct and have it implement error. This way I think we can use reflection to check the error. This would be a pretty big change (I'd argue a breaking change) and require me to change every server type (info, workflow ...)

Reply:
Do you think the `ServerError` path is the way to go ?